be/src/olap/rowset/segment_v2/segment_iterator.cpp

```diff
@@ -1040,6 +1043,55 @@ Status SegmentIterator::_apply_inverted_index() {
             }
         }
     }
+
+    // delete from _common_expr_ctxs_push_down if a MATCH predicate will be removed from _col_predicates
+    // since it's not necessary to eval it any more to avoid index miss, which is added in _normalize_predicate
+    for (auto pred : _col_predicates) {
+        if (pred->type() == PredicateType::MATCH &&
+            std::find(remaining_predicates.begin(), remaining_predicates.end(), pred) ==
```

Comment:
If predicate found in `remaining_predicates` means that, it is not applied by inverted index and then continue expr to filter?

Reply:
yes